paymentsheet/src/main/java/com/stripe/android/lpmfoundations/luxe/LpmRepository.kt

```diff
@@ -152,8 +153,13 @@ class LpmRepository(
         metadata: PaymentMethodMetadata,
         sharedDataSpec: SharedDataSpec,
     ): SupportedPaymentMethod? {
-        return PaymentMethodRegistry.definitionsByCode[sharedDataSpec.type]
-            ?.supportedPaymentMethod(metadata, sharedDataSpec)
+        val paymentMethodDefinition = PaymentMethodRegistry.definitionsByCode[sharedDataSpec.type]
+
+        if (paymentMethodDefinition?.isSupported(metadata) == true) {
```

Comment:
This slightly more strict than it was previously. Now we're filtering out unsupported LPMs here. We were previously doing it for USBank here, but all other LPMs were more or less filtered out in the UI due to restrictions from delayed payment methods and shipping.